File: VulkanGameEngine/InputComponent.cpp
#include "InputComponent.h"
#include "MemoryManager.h"

InputComponent::InputComponent() : GameObjectComponent()
{
}

InputComponent::InputComponent(uint32 gameObjectId) : GameObjectComponent(gameObjectId, kInputComponent)
{
   // transform2DComponentRef = SharedPtr<Transform2DComponent>(CSobject->InvokeMethod<Transform2DComponent*>("GetCPPComponentPtr"));
}

InputComponent::InputComponent(uint32 gameObjectId, const String& name) : GameObjectComponent(gameObjectId, name, kInputComponent)
{
    //auto adfd = this;
    //auto fds4 = static_cast<Transform2DComponent*>(parentGameObjectPtr->GameObjectComponentList[0].get())->GetCPPObjectHandle();
    //auto fds4s = parentGameObjectPtr->GameObjectComponentList[0].get();
    //void* asd = CSobject->InvokeMethod<void*>("GetCPPComponentPtr");
    //Transform2DComponent* fds = static_cast<Transform2DComponent*>(asd);
    //transform2DComponentRef = SharedPtr<Transform2DComponent>(fds);
}

InputComponent::~InputComponent()
{
}

void InputComponent::Input(const float& deltaTime)
{
    GameObjectComponent::Input(deltaTime);
}

void InputComponent::Update(VkCommandBuffer& commandBuffer, const float& deltaTime)
{
    GameObjectComponent::Update(commandBuffer, deltaTime);
}

void InputComponent::Destroy()
{
    GameObjectComponent::Destroy();
}

SharedPtr<GameObjectComponent> InputComponent::Clone() const
{
    return std::make_shared<InputComponent>(*this);
}

size_t InputComponent::GetMemorySize() const
{
    return sizeof(InputComponent);
}

File: VulkanGameEngineLevelEditor/Components/InputComponent.cs
using System.Linq;
using VulkanGameEngineLevelEditor.GameEngineAPI;
using VulkanGameEngineLevelEditor.Vulkan;

namespace VulkanGameEngineLevelEditor.Components
{
    public unsafe class InputComponent : GameObjectComponent
    {
        Transform2DComponent transform;

        public InputComponent() : base()
        {
            Name = "InputComponent";
            ComponentType = ComponentTypeEnum.kInputComponent;
        }

        public InputComponent(GameObject parentGameObject) : base(parentGameObject, ComponentTypeEnum.kInputComponent)
        {
            Name = "InputComponent";
            transform = ParentGameObject.GameObjectComponentList.Where(x => x.ComponentType == ComponentTypeEnum.kTransform2DComponent).First() as Transform2DComponent;
        }

        public InputComponent(GameObject parentGameObject, string name) : base(parentGameObject, name, ComponentTypeEnum.kInputComponent)
        {
            transform = ParentGameObject.GameObjectComponentList.Where(x => x.ComponentType == ComponentTypeEnum.kTransform2DComponent).First() as Transform2DComponent;
        }

        public override void Destroy()
        {

        }

        public override void Draw(VkCommandBuffer commandBuffer, VkPipeline pipeline, VkPipelineLayout pipelineLayout, ListPtr<VkDescriptorSet> descriptorSetList, SceneDataBuffer sceneProperties)
        {

        }

        public override int GetMemorySize()
        {
            return sizeof(InputComponent);
        }

        //public override void Input(KeyBoardKeys key, float deltaTime)
        //{
        //    if (transform != null)
        //    {
        //        if (key == KeyBoardKeys.W)
        //        {
        //            transform.GameObjectPosition.y += 0.01f;
        //        }
        //        if (key == KeyBoardKeys.A)
        //        {
        //            transform.GameObjectPosition.x -= 0.01f;
        //        }
        //        if (key == KeyBoardKeys.S)
        //        {
        //            transform.GameObjectPosition.y -= 0.01f;
        //        }
        //        if (key == KeyBoardKeys.D)
        //        {
        //            transform.GameObjectPosition.x += 0.01f;
        //        }
        //    }
        //    else
        //    {
        //        Console.WriteLine($"Transform not found: GameObjectId: 0x{CPPcomponentPtr.ToString("x")} TransformID: 0x{CPPcomponentPtr.ToString("x")}");
        //    }
        //}

        public void Input(int key, float deltaTime)
        {
            if (transform != null)
            {
                if ((KeyBoardKeys)key == KeyBoardKeys.W)
                {
                    //  transform.GameObjectPosition.y += 0.01f;
                }
                if ((KeyBoardKeys)key == KeyBoardKeys.A)
                {
                    //  transform.GameObjectPosition.x -= 0.01f;
                }
                if ((KeyBoardKeys)key == KeyBoardKeys.S)
                {
                    //  transform.GameObjectPosition.y -= 0.01f;
                }
                if ((KeyBoardKeys)key == KeyBoardKeys.D)
                {
                    //  transform.GameObjectPosition.x += 0.01f;
                }
            }
            else
            {
                //  Console.WriteLine($"Transform not found: GameObjectId: 0x{CPPcomponentPtr.ToString("x")} TransformID: 0x{CPPcomponentPtr.ToString("x")}");
            }
        }

        public override void Update(VkCommandBuffer commandBuffer, float deltaTime)
        {

        }

    }
}
